### PR TITLE
Added pager duty winston transport

### PR DESCRIPTION
**Pull request by chrismaree**
This PR adds integration to Winston to send alerts to Pager duty. Any Winston message with a level of `warn` or `error` will be sent as a pager duty alert. To configure this a number of environment variables are required to be added. The required config is shown within the `.env_sample`. Two example alerts are shown in the screenshot below:
image

### CODE REVIEW

File: core/.env_sample [hunk 1]
```diff
@@ -1,8 +1,13 @@
-MNEMONIC=""           # truffle wallet mnemonic. Used in all scripts, tools & cli.
-SLACK_WEBHOOK=""      # OPTIONAL: webhook to send messages to a slack channel. See Logger.js
+MNEMONIC=""             # truffle wallet mnemonic. Used in all scripts, tools & cli.
+# To enable slack messaging from winston a slack webhook is needed.
+SLACK_WEBHOOK=""        # Webhook to send messages to a slack channel. See Logger.js
```
Comment (nicholaspai) [inline core/.env_sample:3]: Why is this no longer optional?

File: financial-templates-lib/logger/PagerDutyTransport.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  constructor(winstonOpts, pagerDutyOptions) {
+    super(winstonOpts);
+    this.serviceId = pagerDutyOptions.pdServiceId;
+    this.fromEmail = pagerDutyOptions.fromEmail;
+    this.pd = new pdClient(pagerDutyOptions.pdApiToken);
+  }
+
+  async log(info, callback) {
+    // TODO: refactor this processing to better parse mrkdwn and complex data structure from winston.
+    await this.pd.incidents.createIncident(this.fromEmail, {
```
Comment (nicholaspai) [inline financial-templates-lib/logger/PagerDutyTransport.js:16]: Just checking, is this consistent with how you handle errors in the `log()` methods for other transports?

File: core/.env_sample (hunk 1 above)
Comment (chrismaree) [reply core/.env_sample:3]: They are all optional. I removed that comment to just make things a bit more consistent.

File: financial-templates-lib/logger/PagerDutyTransport.js (hunk 2 above)
Comment (chrismaree) [reply financial-templates-lib/logger/PagerDutyTransport.js:16]: Ye so the `log` function gets called whenever the logging event occurs from Winston. This basically means that after the logging is initiated the pager duty library is called.

**Pull request merged**